actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -134,6 +134,8 @@ module ClassMethods
       # * <tt>:exception</tt> - Raises ActionController::InvalidAuthenticityToken exception.
       # * <tt>:reset_session</tt> - Resets the session.
       # * <tt>:null_session</tt> - Provides an empty session during request but doesn't reset it completely. Used as default if <tt>:with</tt> option is not specified.
+      #   Note if <tt>default_protect_from_forgery</tt> is true, Rails call protect_from_forgery with <tt>with :exception</tt>.
+      #   This might not be intuitive as the method itself treats <tt>:with</tt> as <tt>:null_session</tt> by default.
```

Comment:
I think putting

> This might not be intuitive

in the docs is not a great look. We do already mention that `:null_session` is the default value, but maybe that should be moved up to the `:with` line itself instead of here?

I do think mentioning the configuration is a good idea, but maybe it could be mentioned much earlier instead here. My thought process is: if a user looks up this method, the fact that Rails applies this to `ActionController::Base` by default is probably important enough that it shouldn't be a Note buried in the list of options.

Reply:
Thank you. I made a change with f7cc8b31913fe25a6a348536292cec00582ee6b5.